infra/discovery/src/lib/scoring.js

```diff
@@ -0,0 +1,56 @@
+async function scoreListing(listing) {
+  let score = 1.0
+
+  // Downrank anything not active (so lower pending / sold / withdrawn)
+  if (listing.status !== 'active') {
+    score *= 0.3
+  }
+
+  // Downrank "cheap" listings to hide wash/spam transactions.
+  // We might want to make this a smoother transition later.
+  // ETH value may need to become a dynamic parameter later.
+  const cheapListingThresholds = {
+    'fiat-USD': 1.5,
+    'fiat-EUR': 1.35,
+    'token-DAI': 1.5,
+    'token-ETH': 0.01
+  }
+  if (
+    listing.price &&
+    listing.price.amount &&
+    listing.price.currency &&
+    listing.price.currency.id
+  ) {
+    const cheapThreshold = cheapListingThresholds[listing.price.currency.id]
+    if (cheapThreshold) {
+      if (listing.price.amount <= cheapThreshold) {
```

Comment:
I think listing.price.amount is returned as a string.
Does this comparison between a string and a float work in JS ?
Perhaps let's explicitly cast to float ?

Reply:
Yeah, this works, but an explicit cast will be better.